diffparser.go

```diff
@@ -36,6 +38,24 @@ type DiffRange struct {
 	Lines []*DiffLine
 }
 
+func retry(ctx context.Context, f func() error) (int, error) {
+	attempt := 0
+	for delay := time.Millisecond; ; delay *= 2 {
+		attempt++
+		if err := f(); err == nil {
+			return attempt, nil
+		}
+		if delay >= 10*time.Second {
+			delay = 10 * time.Second
+		}
+		select {
+		case <-time.After(delay):
```

Comment:
This is going to create a new time.After object on each iteration and leak mem for the duration of the for loop.

Reply:
@codelingobot capture Ticker in for select

Make one ticker as opposed to making a new timer each iteration.

Reply:
@codelingo capture test rule

Reply:
@codelingo capture Ticker in for select

Make one ticker as opposed to making a new timer each on iteration.

Reply:
@codelingobot capture Ticker in for select

Make one ticker as opposed to making a new timer each on iteration.

Reply:
@CodeLingoBot capture Ticker in for select

Make one ticker as opposed to making a new timer each on iteration.